File: Python/1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashTable = {} # hash saves index of the element
        for index, elem in enumerate(nums):
            needNum = target - elem # we need this num to have target
            # it is in case needNum == elem -> to ensure have two of the same key
            # need to check it is not the same one by index
            if (needNum in hashTable) and (hashTable[needNum] != index):
                # we can return as the q is about two sums
                return list((index, hashTable[needNum]))
            hashTable[elem] = index

File: CPP/1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        // If there is only one valid answer, and two numbers;
        // Target - currentNum = needNum -> Check if I have needNum
        // Since we have to scan if the num exsits every loop -> choose map to have O(lg n)
        map<int, int> table;

        for (int index = 0; index < nums.size(); ++index){
            int needNum = target - nums[index];
            if (table.find(needNum) != table.end()){
                if (table[needNum] != index){
                    cout << needNum << nums[index] << endl;
                    return {table[needNum], index};
                }
            }
            table[nums[index]] = index;
        }
        return {0,0};
    }
};
// Implementation: To avoid returning the same two indices, Compare value with
// ONly what is in the table (previously checked numbers)
// That is why in Line 14: index is the second.
